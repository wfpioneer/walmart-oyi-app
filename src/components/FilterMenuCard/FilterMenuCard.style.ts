import { StyleSheet } from 'react-native';
import COLOR from '../../themes/Color';

export default StyleSheet.create({
  menuCardText: {
    flex: 1,
    justifyContent: 'center'
  },
  subtitleText: {
    color: COLOR.GREY_500,
    fontSize: 12
  },
  arrowView: {
    flex: 0.1
  }
});
