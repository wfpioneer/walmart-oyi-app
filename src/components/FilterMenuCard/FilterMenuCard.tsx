import React from 'react';
import { Text, View } from 'react-native';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import COLOR from '../../themes/Color';
import styles from './FilterMenuCard.style';

interface MenuCardProps {
    title: string;
    subtext: string;
    opened: boolean;
  }

export const MenuCard = (props: MenuCardProps): JSX.Element => {
  const iconName = props.opened ? 'keyboard-arrow-up' : 'keyboard-arrow-down';
  return (
    <>
      <View style={styles.menuCardText}>
        <Text>{props.title}</Text>
        <Text style={styles.subtitleText}>{props.subtext}</Text>
      </View>
      <View style={styles.arrowView}>
        <MaterialIcons name={iconName} size={25} color={COLOR.GREY_700} />
      </View>
    </>
  );
};
