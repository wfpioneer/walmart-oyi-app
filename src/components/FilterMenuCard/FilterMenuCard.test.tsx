import { render } from '@testing-library/react-native';
import React from 'react';
import { MenuCard } from './FilterMenuCard';

describe('Filter menu card render tests', () => {
  it('Renders MenuCard with the dropdown icon Open', () => {
    const { toJSON } = render(
      <MenuCard
        title="Menu Card Title"
        subtext="Sub Text Opened"
        opened={true}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('Renders MenuCard with the dropdown icon Closed', () => {
    const { toJSON } = render(
      <MenuCard
        title="Menu Card Title"
        subtext="Sub Text Closed"
        opened={false}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
