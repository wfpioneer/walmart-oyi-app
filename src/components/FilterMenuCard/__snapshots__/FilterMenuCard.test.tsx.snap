// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter menu card render tests Renders MenuCard with the dropdown icon Closed 1`] = `
Array [
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text>
      Menu Card Title
    </Text>
    <Text
      style={
        Object {
          "color": "#9e9e9e",
          "fontSize": 12,
        }
      }
    >
      Sub Text Closed
    </Text>
  </View>,
  <View
    style={
      Object {
        "flex": 0.1,
      }
    }
  >
    <Icon
      color="#616161"
      name="keyboard-arrow-down"
      size={25}
    />
  </View>,
]
`;

exports[`Filter menu card render tests Renders MenuCard with the dropdown icon Open 1`] = `
Array [
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text>
      Menu Card Title
    </Text>
    <Text
      style={
        Object {
          "color": "#9e9e9e",
          "fontSize": 12,
        }
      }
    >
      Sub Text Opened
    </Text>
  </View>,
  <View
    style={
      Object {
        "flex": 0.1,
      }
    }
  >
    <Icon
      color="#616161"
      name="keyboard-arrow-up"
      size={25}
    />
  </View>,
]
`;
